src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const initialCompareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      }
+
+      return 1;
```

Comment:
The initialCompareFunction does not handle the case when `a` is equal to `b`. In such a case, it should return 0 to indicate that no sorting is required for equal elements.